improver/utilities/spatial.py

```diff
@@ -99,11 +101,22 @@ def calculate_grid_spacing(
     coord = cube.coord(axis=axis).copy()
     coord.convert_units(units)
     diffs = np.unique(np.diff(coord.points))
+
     if len(diffs) > 1:
-        raise ValueError(
-            "Coordinate {} points are not equally spaced".format(coord.name())
-        )
-    return diffs[0]
+        diffs_diff = np.diff(diffs)
+        diffs_mean = np.mean(diffs)
+
+        if np.any(abs(diffs_diff) > rtol * diffs_mean):
```

Comment:
Just looking through to refresh my numpy! If you have negative diffs, would you get a problem from the abs/mean?